helix-core/src/test/java/org/apache/helix/integration/rebalancer/TestAutoRebalance.java

```diff
@@ -229,6 +236,16 @@ static boolean verifyBalanceExternalView(ZNRecord externalView, int partitionCou
     return partitionCount == totalCount;
   }
 
+  private void setupAutoRebalancer() {
+    HelixAdmin admin = _gSetupTool.getClusterManagementTool();
```

Comment:
This actually relies on the fact that 

```
 _gSetupTool.addResourceToCluster(CLUSTER_NAME, "MyDB2", _PARTITIONS, "MasterSlave",
        RebalanceMode.FULL_AUTO + "");
```

would add  idea state using DelayedAutoRebalancer + AutoRebalance strategy

here, we change the idea state to use autoRebalancer, thus eventually we end up with autoRebalancer + autoRebalance strategy.

The thing is that if later people change the gSetupTool.addResourceToCluster() similar to testhelper to have DelayedAutoRebalancer + crushEd, this test won't work. 

Shall we add a comment that we depends on  _gSetupTool.addResourceToCluster to use default AutoRebalanceStrategy, or just note here we need autoRebalancer + autoRebalance strategy.

Reply:
This test is for testing AutoRebalancer, so no matter what's the default behavior change, it is the right test logic. If it fails, then the new code change shall be fixed. I think there is no necessity of adding comment. Because this change only makes the logic more explicit.